pkg/storage/stores/chunk/chunkstore.go

```diff
@@ -0,0 +1,40 @@
+package chunk
+
+import (
+	"context"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/loki/pkg/iter"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/logql"
+	"github.com/grafana/loki/pkg/storage/chunk"
+	"github.com/grafana/loki/pkg/storage/chunk/fetcher"
+)
+
+// Reader is an interface for a chunk store that reads data
+type Reader interface {
```

Comment:
i feel that `Reader` is not exactly a reader for chunks
looking at the code refs, should we move this to querier instead. wdyt?

Reply:
I agree, `Reader` is probably not the best name for what the interface defines.

Reply:
`Reader -> ReadStore`